os/mm/mm_heap/mm_getheap.c

```diff
@@ -40,25 +45,26 @@ extern struct mm_heap_s g_mmheap[CONFIG_MM_NHEAPS];
  ****************************************************************************/
 struct mm_heap_s *mm_get_heap(void *address)
 {
-	int heap_idx;
 #ifdef CONFIG_MM_KERNEL_HEAP
-	struct tcb_s *tcb;
-
-	tcb = sched_gettcb(getpid());
-	if (tcb->flags & TCB_FLAG_TTYPE_MASK == TCB_FLAG_TTYPE_KERNEL) {
-		return &g_kmmheap;
-	} else
+	if (address > (FAR void *)g_kmmheap.mm_heapstart[0] && address < (FAR void *)g_kmmheap.mm_heapend[0]) {
```

Comment:
Are the boundary values invalid for address check ?

Reply:
I think the boundary values are valid. If kernel heap is defined, we just check if the address falls within the kernel heap.

Reply:
>= ?

Reply:
Okay. Done.